From 31ebafdd29c9f0a90d4fc4771992d1e2c605a65d Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Mon, 30 May 2022 13:44:08 +0200
Subject: [PATCH 037/120] arm: stm32mp: add support of TAMP_COPRO backup
 registers on stm32mp2

Add definition to support the backup registers in U-boot.
this first implementation is based on a fixed mapping of
the backup-registers.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git 569a2ad6d0e3279a95fc1124b4b259f9bc512c93 arm: stm32mp: add support of TAMP_COPRO backup registers on stm32mp2]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I400fff5626c34684d192cc5d3fa07969f4361cef
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/315247
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Michel JAOUEN <michel.jaouen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm/mach-stm32mp/include/mach/stm32.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/mach-stm32mp/include/mach/stm32.h b/arch/arm/mach-stm32mp/include/mach/stm32.h
index a9ac49bc5d2..41eed82cfa5 100644
--- a/arch/arm/mach-stm32mp/include/mach/stm32.h
+++ b/arch/arm/mach-stm32mp/include/mach/stm32.h
@@ -183,8 +183,13 @@ enum forced_boot_mode {
 /* TAMP registers x = 0 to 127 : hardcoded description, waiting NVMEM node in DT */
 #define TAMP_BACKUP_REGISTER(x)		(STM32_TAMP_BASE + 0x100 + 4 * (x))
 
+/* TAMP registers zone 2 RIF 2 (RO) at 72 */
+#define TAMP_COPRO_STATE TAMP_BACKUP_REGISTER(72)
+
 /* TAMP registers zone 3 RIF 1 (RW) at 96*/
 #define TAMP_BOOT_CONTEXT		TAMP_BACKUP_REGISTER(96)
+#define TAMP_COPRO_RSC_TBL_ADDRESS	TAMP_BACKUP_REGISTER(97)
+#define TAMP_COPRO_RSC_TBL_SIZE		TAMP_BACKUP_REGISTER(98)
 #endif /* STM32MP25X */
 
 /* offset used for BSEC driver: misc_read and misc_write */
-- 
2.47.2

